modules/planning/open_space/coarse_trajectory_generator/grid_search.cc

```diff
@@ -57,6 +59,48 @@ bool GridSearch::CheckConstraints(std::shared_ptr<Node2d> node) {
 
 std::vector<std::shared_ptr<Node2d>> GridSearch::GenerateNextNodes(
     std::shared_ptr<Node2d> current_node) {
+  double current_node_x = current_node->GetGridX();
+  double current_node_y = current_node->GetGridY();
+  double current_node_path_cost = current_node->GetPathCost();
+  double diagonal_distance = std::sqrt(2.0);
+  std::vector<std::shared_ptr<Node2d>> next_nodes;
+  std::shared_ptr<Node2d> up =
+      std::make_shared<Node2d>(current_node_x, current_node_y + 1.0, XYbounds_);
+  up->SetPathCost(current_node_path_cost + 1.0);
+  std::shared_ptr<Node2d> up_right = std::make_shared<Node2d>(
+      current_node_x + 1.0, current_node_y + 1.0, XYbounds_);
+  up_right->SetPathCost(current_node_path_cost + diagonal_distance);
+  std::shared_ptr<Node2d> right =
+      std::make_shared<Node2d>(current_node_x + 1.0, current_node_y, XYbounds_);
+  right->SetPathCost(current_node_path_cost + 1.0);
+  std::shared_ptr<Node2d> down_right = std::make_shared<Node2d>(
+      current_node_x + 1.0, current_node_y - 1.0, XYbounds_);
+  down_right->SetPathCost(current_node_path_cost + diagonal_distance);
+  std::shared_ptr<Node2d> down =
+      std::make_shared<Node2d>(current_node_x, current_node_y - 1.0, XYbounds_);
+  down->SetPathCost(current_node_path_cost + 1.0);
+  std::shared_ptr<Node2d> down_left = std::make_shared<Node2d>(
+      current_node_x - 1.0, current_node_y - 1.0, XYbounds_);
+  down_left->SetPathCost(current_node_path_cost + diagonal_distance);
+  std::shared_ptr<Node2d> left =
+      std::make_shared<Node2d>(current_node_x - 1.0, current_node_y, XYbounds_);
+  left->SetPathCost(current_node_path_cost + 1.0);
+  std::shared_ptr<Node2d> up_left = std::make_shared<Node2d>(
+      current_node_x - 1.0, current_node_y + 1.0, XYbounds_);
+  up_left->SetPathCost(current_node_path_cost + diagonal_distance);
+
+  next_nodes.emplace_back(up);
+  next_nodes.emplace_back(up_right);
+  next_nodes.emplace_back(right);
+  next_nodes.emplace_back(down_right);
+  next_nodes.emplace_back(down);
+  next_nodes.emplace_back(down_left);
+  next_nodes.emplace_back(left);
+  next_nodes.emplace_back(up_left);
+  return next_nodes;
+}
+/*std::vector<std::shared_ptr<Node2d>> GridSearch::GenerateNextNodes(
+    std::shared_ptr<Node2d> current_node) {
```

Comment:
Just remove the paragraph if the function is no longer needed.

Besides, prefer C++ style comment (`//`) to c-style `/* ... */`.